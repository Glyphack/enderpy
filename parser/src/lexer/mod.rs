mod diagnostics;
pub mod fstring;
pub mod lexer;
