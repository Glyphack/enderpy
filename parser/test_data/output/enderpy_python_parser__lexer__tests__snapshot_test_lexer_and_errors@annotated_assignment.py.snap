---
source: parser/src/lexer/mod.rs
description: "a: int = 1\n\nb: tuple[int, int] = 1, 2\n\nc: tuple[str, int] = \"1\", 2, \n\nd: tuple[int, int] = b\n\ne: tuple[int, int]\n\nf: tuple[int, int] = e\n\ng: tuple[int,\n         int] = 1, 2\n\n"
input_file: parser/test_data/inputs/annotated_assignment.py
---
0,1: Identifier                  (Str("a"))
1,2: :                                        (None)
3,6: Identifier                (Str("int"))
7,8: =                                        (None)
9,10: Integer                    (Number("1"))
10,11: NewLine                          (None)
11,12: NewLine                          (None)
12,13: Identifier                (Str("b"))
13,14: :                                      (None)
15,20: Identifier            (Str("tuple"))
20,21: [                                      (None)
21,24: Identifier              (Str("int"))
24,25: ,                                      (None)
26,29: Identifier              (Str("int"))
29,30: ]                                      (None)
31,32: =                                      (None)
33,34: Integer                   (Number("1"))
34,35: ,                                      (None)
36,37: Integer                   (Number("2"))
37,38: NewLine                          (None)
38,39: NewLine                          (None)
39,40: Identifier                (Str("c"))
40,41: :                                      (None)
42,47: Identifier            (Str("tuple"))
47,48: [                                      (None)
48,51: Identifier              (Str("str"))
51,52: ,                                      (None)
53,56: Identifier              (Str("int"))
56,57: ]                                      (None)
58,59: =                                      (None)
60,63: StringLiteral      (Str("\"1\""))
63,64: ,                                      (None)
65,66: Integer                   (Number("2"))
66,67: ,                                      (None)
68,69: NewLine                          (None)
69,70: NewLine                          (None)
70,71: Identifier                (Str("d"))
71,72: :                                      (None)
73,78: Identifier            (Str("tuple"))
78,79: [                                      (None)
79,82: Identifier              (Str("int"))
82,83: ,                                      (None)
84,87: Identifier              (Str("int"))
87,88: ]                                      (None)
89,90: =                                      (None)
91,92: Identifier                (Str("b"))
92,93: NewLine                          (None)
93,94: NewLine                          (None)
94,95: Identifier                (Str("e"))
95,96: :                                      (None)
97,102: Identifier           (Str("tuple"))
102,103: [                                    (None)
103,106: Identifier            (Str("int"))
106,107: ,                                    (None)
108,111: Identifier            (Str("int"))
111,112: ]                                    (None)
112,113: NewLine                        (None)
113,114: NewLine                        (None)
114,115: Identifier              (Str("f"))
115,116: :                                    (None)
117,122: Identifier          (Str("tuple"))
122,123: [                                    (None)
123,126: Identifier            (Str("int"))
126,127: ,                                    (None)
128,131: Identifier            (Str("int"))
131,132: ]                                    (None)
133,134: =                                    (None)
135,136: Identifier              (Str("e"))
136,137: NewLine                        (None)
137,138: NewLine                        (None)
138,139: Identifier              (Str("g"))
139,140: :                                    (None)
141,146: Identifier          (Str("tuple"))
146,147: [                                    (None)
147,150: Identifier            (Str("int"))
150,151: ,                                    (None)
151,152: NewLine                        (None)
161,164: Identifier            (Str("int"))
164,165: ]                                    (None)
166,167: =                                    (None)
168,169: Integer                 (Number("1"))
169,170: ,                                    (None)
171,172: Integer                 (Number("2"))
172,173: NewLine                        (None)
173,174: NewLine                        (None)
